restricted_site_access.php

```diff
@@ -1899,6 +1903,43 @@ public static function get_ip_protocol( $ip = '' ) {
 
 		return false;
 	}
+
+	/**
+	 * Filters custom trusted headers.
+	 *
+	 * All headers should be present in request and have the correct value.
+	 * Example:
+	 * add_filter(
+	 *     'rsa_custom_trusted_headers',
+	 *     function( $headers ) {
+	 *         $headers['x-gt-matrix'] = 'SomeValue1';
+	 *         $headers['x-custom-header'] = 'SomeValue2';
+	 *         return $headers;
+	 *     }
+	 * )
+	 *
+	 * @since x.x.x
+	 */
+	public static function has_valid_custom_header(): ?bool {
+		$allowed_custom_trusted_headers = apply_filters( 'rsa_custom_trusted_headers', array() );
+
+		if ( $allowed_custom_trusted_headers ) {
```

Comment:
We should probably be a little more defensive in this check in case someone uses the above filter incorrectly (for instance, returns a string instead of an array). Something like:

```suggestion
		if ( is_array(  $allowed_custom_trusted_headers ) && ! empty(  $allowed_custom_trusted_headers ) ) {
```